src/internal/Subscriber.ts

```diff
@@ -186,6 +193,19 @@ function defaultErrorHandler(err: any) {
   reportUnhandledError(err);
 }
 
+/**
+ * A handler for notifications that cannot be sent to a stopped subscriber.
+ * @param notification The notification being sent
+ * @param subscriber The stopped subscriber
+ */
+function handleStoppedNotification(
+  notification: { kind: 'N'; value: any } | { kind: 'E'; error: any } | { kind: 'C' },
+  subscriber: Subscriber<any>
+) {
+  const { onStoppedNotification } = config;
+  onStoppedNotification && setTimeout(() => onStoppedNotification(notification, subscriber));
```

Comment:
Q: Should we use a "setTimeoutProvider" here? Do we want this to be testable with the `TestScheduler`? (I guess the same could go for `hostReportError` etc)

Reply:
I've changed the two implementations to use the `immediateProvider`, as we don't have a provider for `setTimeout` - the scheduling implementation uses `setInterval` and using `setImmediate` here didn't involve any keeping-the-return-value-around shenanigans.

The tests still perform their waiting using `setTimeout` 'cause macro tasks happen after micro tasks and 'cause I'm lazy. I can change those, too, if you want.